### ISSUE TITLE
Node crash

### ISSUE THREAD

**Issue by sebastien-forestier** [massalabs/massa#2522]
One of our bootstrap nodes crashed :
[logs_bug.tar.gz](https://github.com/massalabs/massa/files/8457585/logs_bug.tar.gz)

**Comment by damip** [comment]
Seems to be problems around Nodes/Network.

`Failed to send NetworkEvent due to timeout: ReceivedEndorsements `

**Comment by gterzian** [comment]
I see a lot of `[2m2022-04-09T18:05:35.286054Z[0m [34mDEBUG[0m [2mmassa_protocol_worker::worker_operations_impl[0m[2m:[0m re-ask operation vCPYGxTQwx8ESdntTV6a2qZ1r72Bh7yPWRwrHg29hyEvAQF9 asked for the first time 37734 millis ago.`

and apparently not a single `NetworkCommand::SendOperationAnnouncements` receipt being traced(maybe it's because of the log level?). 

I also see the node worker logging it's inablity to send events(to the network worker).

And then we see a lot of failures to send events by the network worker(many seem to be operations announcements coming from other nodes).

**Comment by gterzian** [comment]
so I think one potential improve is not to immediately ask for operations when receiving announcements(same goes for other objects). 

Currently [`on_operations_announcements_received`](https://github.com/massalabs/massa/blob/0835a6175207832d4a7deba00433736b9caa0628/massa-protocol-worker/src/worker_operations_impl.rs#L62) is called both when [receiving announcements](https://github.com/massalabs/massa/blob/0835a6175207832d4a7deba00433736b9caa0628/massa-protocol-worker/src/protocol_worker.rs#L1345), and as part of [`update_ask_operation`](https://github.com/massalabs/massa/blob/0835a6175207832d4a7deba00433736b9caa0628/massa-protocol-worker/src/worker_operations_impl.rs#L176)(which runs according to a timer). 

So one problem might be sending a message to the network worker as part of handling an network event, which when the channels are filled, means each time we have to wait for the network worker to drop an event before it can handle the command. 

I propose that when receiving announcement, we add them to `op_batch_buffer`, and then process as part of `update_ask_operation` at the next timer.

That way when the network sends us an event, it is handled without sending a command back to the network worker, which means that protocol goes back to its select loop and is able to handle the next event from network. 

Eventually, the network commands are sent, but only as part when the timer hits. 

Also, since we update the ask operations state is updated(via `checked_operations`) when receiving new operations via network events, it is possible that by only asking for operations via the timer, instead of immediately when receiving the announcements, we end-up asking for less operations since we may already receive it in the meantime. 

This would require breaking up `on_operations_announcements_received` into a "buffer asks based on the announcement" part, which would be called when receiving announcements, and a "send ask for operations" part, which would be called by `update_ask_operations`. 

cc @adrien-zinger

**Comment by damip** [comment]
Here are the logs of the `add_debug_logs` branch. It has logs:
* in consensus, protocol, network
  * before every select! in main loops
  * at the beginning of every select! branch
  * at the end of the select!

https://drive.google.com/file/d/1BsVnmq1WD21KsE1gW1Sub6rpZBzbK2uo/view?usp=sharing

The goal is to see if a loop was saturated or blocked

**Comment by gterzian** [comment]
Currently trying out a few things based on https://github.com/massalabs/massa/issues/2522#issuecomment-1094772283

**Comment by adrien-zinger** [comment]
Hi @gterzian, here is a [link](https://htmlpreview.github.io/?https://github.com/adrien-zinger/operation_propagation_presentation/blob/main/index.html) of the main algorithm presentation. There is a part about the corner case.

About the channel's obstruction, we potentially double the event's with announcements + operations, in the latest feature.
Solutions can be any management of event limitations:

- At the creation  of announcement (limited by seconds).
- At the creation of asking batches (in the re-trigger operations loop).
- Just double the channel buffer.

**Comment by gterzian** [comment]
Still tweaking a few things at https://github.com/massalabs/massa/pull/2527

Could already try to run a test with that branch?

@adrien-zinger I can't find the part on corner cases in the presentation, which slide should it be?

**Comment by gterzian** [comment]
Addressing the above in https://github.com/massalabs/massa/pull/2528

**Comment by gterzian** [comment]
> you think that the channels between pool and protocols are also full

Yes and also that we are probably broadcasting too many operations, since currently we ask for operations from pool for a given `node_id`, and then when receiving the operations we immediately send them to the `node_id`, without checking whether the node might not be interested anymore(for example if the node broadcasted announcements for those operations).

**Comment by Punkoivan** [comment]
The issue is still here, I have similar logs:
```
2022-05-14T10:33:42.057816Z  WARN massa_protocol_worker::protocol_worker: Failed to send ProtocolPoolEvent due to timeout: ReceivedOperations { operations: {2rEzYUziWuGpiSM7CqeBiQSPQjHaXbRfw697DfpWDV2AFdeMbi: Signed { content: Operation { sender_public_key: PublicKey(PublicKey(dbc7a5fe6e6abc85e5e8ad17d8691fdf2d35d3abad49d438b550254913eb4cf7fbc6749bc7628f0a2c09dd3746db030350e9e4a7783368b35cedb434e4689c08)), fee: Amount(0), expire_period: 45584, op: Transaction { recipient_address: Address(2fYKuDCS9y6oLRZT5Vc8gBFXPzYZibUNbPREb19U42xLKcjoqa), amount: Amount(140136) } }, signature: Signature(304402205aab358513432e1a1570495ca69be32fad22ab4963e300841456d059de59f08202201ff62095f4cfe5f9f8b58443ab01e6142be398ba7f65150b520f1a2cd1831659), phantom: PhantomData }, ayKbNTz33R3fA3WyxJdezhk9G7YUT5QoTAzsaGeXVS7twpCDE: Signed { content: Operation { sender_public_key: PublicKey(PublicKey(dbc7a5fe6e6abc85e5e8ad17d8691fdf2d35d3abad49d438b550254913eb4cf7fbc6749bc7628f0a2c09dd3746db030350e9e4a7783368b35cedb434e4689c08)), fee: Amount(0), expire_period: 45584, op: Transaction { recipient_address: Address(2fYKuDCS9y6oLRZT5Vc8gBFXPzYZibUNbPREb19U42xLKcjoqa), amount: Amount(140108) } }, signature: Signature(3045022100eb650a4f8d15f4010d59ff711422a1818d9942e64c81679759ea9dfde674d14502204e3552525b3314f1336f3cf224f8b1789a50845ba822e5ede0087df28394d29a), phantom: PhantomData }, zACNEPG3fD8gFsLZyE55iRUZed64Q7uhrCDSJMNMh58aRpdU2: Signed { content: Operation { sender_public_key: PublicKey(PublicKey(dbc7a5fe6e6abc85e5e8ad17d8691fdf2d35d3abad49d438b550254913eb4cf7fbc6749bc7628f0a2c09dd3746db030350e9e4a7783368b35cedb434e4689c08)), fee: Amount(0), expire_period: 45584, op: Transaction { recipient_address: Address(2fYKuDCS9y6oLRZT5Vc8gBFXPzYZibUNbPREb19U42xLKcjoqa), amount: Amount(140177) } }, signature: Signature(304402204842c3d5bc9827e818a500d02989fc059cfe5ed613575ecd9808dba12bc71a6d022077198acc4a0bc9c77570e460d0863a1fdc36222e9840dab4c2af56bb11dec5f2), phantom: PhantomData }}, propagate: true }.
```

Node offline for a 5 days of TEST 10.

And looks like noone have a clue what to do.

I've tried recompile, starting from scratch, ipv6 - no luck.
